app/Http/Controllers/Backend/User/DashboardController.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace App\Http\Controllers\Backend\User;
+
+use App\Enum\StatusVisibility;
+use App\Http\Controllers\Controller;
+use App\Models\Status;
+use App\Models\User;
+use Carbon\Carbon;
+use Illuminate\Contracts\Pagination\Paginator;
+use Illuminate\Database\Eloquent\Builder;
+
+abstract class DashboardController extends Controller
+{
+
+    public static function getPrivateDashboard(User $user): Paginator {
+        $followingIDs   = $user->follows->pluck('id');
+        $followingIDs[] = $user->id;
+        return Status::with([
+                                'event', 'likes', 'user', 'trainCheckin',
+                                'trainCheckin.Origin', 'trainCheckin.Destination',
+                                'trainCheckin.HafasTrip.stopoversNEW.trainStation'
+                            ])
+                     ->join('train_checkins', 'train_checkins.status_id', '=', 'statuses.id')
+                     ->select('statuses.*')
+                     ->where('train_checkins.departure', '<', Carbon::now()->addMinutes(20)->toIso8601String())
+                     ->orderBy('train_checkins.departure', 'desc')
+                     ->whereIn('statuses.user_id', $followingIDs)
+                     ->whereIn('visibility', [StatusVisibility::PUBLIC, StatusVisibility::FOLLOWERS])
+                     ->orWhere('statuses.user_id', $user->id)
+                     ->withCount('likes')
+                     ->latest()
+                     ->simplePaginate(15);
+    }
+
+    public static function getGlobalDashboard(User $user): Paginator {
```

Comment:
Do we really need to supply a user here? Isn't a default `auth()->user()` enough?

Reply:
Each user has a different global dashboard based on private profiles and their follows. I think it makes sense to differentiate explicitly.

Reply:
In addition, you can theoretically test better this way.